### PR TITLE
Work around broken libc++ array implementation

### PR DESCRIPTION
**Pull request by wthrowe**
### Types of changes:

- [ ] Bugfix
- [ ] New feature

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests, `clang-tidy` and `IWYU`. For
  instructions on how to perform the CI checks locally refer to the [Dev guide
  on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run `make doc`
  to generate the documentation locally into `BUILD_DIR/docs/html`. Then open
  `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

**Review by kidder — DISMISSED** [review]
@nilsdeppe please look at

File: tests/Unit/Utilities/Test_MakeArray.cpp [hunk 1]
```diff
@@ -111,6 +111,7 @@ SPECTRE_TEST_CASE("Unit.Utilities.MakeArray", "[Unit][Utilities]") {
   // Check make_array with an rvalue sequence
```
Comment (nilsdeppe) [inline tests/Unit/Utilities/Test_MakeArray.cpp:111]: I don't understand the commit message.

File: CMakeLists.txt [hunk 2]
```diff
@@ -40,6 +40,8 @@ include(SetupSanitizers)
 include(SetupListTargets)
```
Comment (nilsdeppe) [inline CMakeLists.txt:40]: The commit message should both describe the bug and link to the libc++ bug.

**Review by nilsdeppe — CHANGES_REQUESTED** [review]

File: tests/Unit/Utilities/Test_MakeArray.cpp (hunk 1 above)
Comment (kidder) [reply tests/Unit/Utilities/Test_MakeArray.cpp:111]: maybe "address clang-tidy complaint"

**Review by kidder — APPROVED** [review]
@nilsdeppe please look at

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**